### PR TITLE
chore(icon): accessibility checklist

### PR DESCRIPTION
**Pull request by 321gillian**
✅ Closes: COMUI-3328

### CODE REVIEW

File: packages/genesys-spark/src/hosts.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function getChartComponentAssetsOrigin(): string {
   if (IS_DEV_MODE == true) {
     // This conditional is optimized out in production due to @rollup/plugin-replace
     // and rollup's dead code elimination
-    return 'http://localhost:3734';
+    return `http://${window.location.hostname}:3734`;
```
Comment (321gillian) [inline packages/genesys-spark/src/hosts.ts:40]: This isn't related to accessibility but I've been testing some stuff through Parallels on a Windows machine and this makes port forwarding easier.

File: packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.tsx [hunk 2]
```diff
@@ -100,6 +100,13 @@ export class GuxIcon {
       svgElement.setAttribute('aria-hidden', String(this.decorative));
     } else {
       svgElement.setAttribute('aria-hidden', 'false');
+
+      /*
+        If the icon is not decorative, we need to set the role to img to ensure that
+        the screenreader will announce the localized screenreader-text property.
+        https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg
+      */
+      svgElement.setAttribute('role', 'img');
```
Comment (321gillian) [inline packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.tsx:109]: NVDA was reading the icons with screenreader text as blank so this change fixes that.

Comment (321gillian) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.tsx:109]: In Firefox, it's down as a 'graphics-document'

Comment (321gillian) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.tsx:109]: @daragh-king-genesys happy to move this part to a separate issue in the backlog if you'd prefer

Comment (daragh-king-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.tsx:109]: Ya move it into a seperate issue

**Comment by 321gillian** [comment]
Added a follow-on ticket here but I'm not sure that it merits failing the component since it works generally https://inindca.atlassian.net/browse/COMUI-3359

**Review by jordanstith15 — APPROVED** [review]

**Pull request merged**